src/common/commands.ts

```diff
@@ -36,3 +38,86 @@ async function executeCommand(lsClient: LanguageClient, command: string) {
     );
   });
 }
+
+/**
+ * Creates a debug information provider for the `ruff.printDebugInformation` command.
+ *
+ * This will open a new editor window with the debug information considering the active editor.
+ */
+export function createDebugInformationProvider(
+  getClient: () => LanguageClient | undefined,
+  serverId: string,
+  context: vscode.ExtensionContext,
+) {
+  let configuration = getConfiguration(serverId) as unknown as ISettings;
+  if (configuration.nativeServer === false || configuration.nativeServer === "off") {
+    return async () => {
+      vscode.window.showInformationMessage(
+        "Debug information is only available when using the native server",
+      );
+    };
+  }
+
+  const contentProvider = new (class implements vscode.TextDocumentContentProvider {
+    readonly uri = vscode.Uri.parse("ruff-server-debug://debug");
+    readonly eventEmitter = new vscode.EventEmitter<vscode.Uri>();
+
+    async provideTextDocumentContent(_uri: vscode.Uri): Promise<string> {
+      const lsClient = getClient();
+      if (!lsClient) {
+        return "";
+      }
+      const textEditor = vscode.window.activeTextEditor;
+      const notebookEditor = vscode.window.activeNotebookEditor;
+      const params = {
+        command: `${serverId}.printDebugInformation`,
+        arguments: [
+          {
+            textDocument: notebookEditor
+              ? { uri: notebookEditor.notebook.uri.toString() }
+              : textEditor
+              ? { uri: textEditor.document.uri.toString() }
+              : undefined,
+          },
+        ],
+      };
```

Comment:
I updated this in this PR from `main` to check if we're in a notebook context and then use text document. I don't think there's any specific information for a notebook cell so using a notebook for them should be fine.